src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +120,309 @@ class boss_gluth : public CreatureScript
+                timer += diff;
+                Creature* gluth = ObjectAccessor::GetCreature(*me, GluthGUID);
+                // Putting this in the UpdateAI loop fixes an issue where death gripping a decimated zombie would make the zombie stand still until the rest of the fight.
+                // Also fix the issue where if one or more zombie is rooted when decimates hits (and MovePoint() is called), the zombie teleport to the boss. pretty weird behavior.
+                if (me->GetExactDist2d(gluth) > 10.0 && me->CanFreeMove() && timer>1600) // it takes about 1600 ms for the animation to cycle. This way, the animation looks relatively smooth.
                 {
-                    me->Kill(me->GetVictim());
-                    me->ModifyHealth(int32(me->CountPctFromMaxHealth(5)));
+                    me->GetMotionMaster()->MovePoint(0, gluth->GetPosition()); // isn't dynamic. So, to take into account Gluth's movement, it must be called periodicly.
+                    //me->GetMotionMaster()->MoveFollow(gluth, 0, 0); // the zombies will stand still too far away from Gluth at the end of their movement. Also doesnt seem to work with SetWalk(true)
+                    timer = 0;
                 }
+
+                if (me->GetExactDist2d(gluth) <= 10.0)
+                    me->StopMoving();
             }
-            else
+            else if (state == STATE_ZOMBIE_NORMAL)
                 DoMeleeAttackIfReady();
         }
+
+        void SetData(uint32 id, uint32 value) override
+        {
+            if (id == DATA_ZOMBIE_STATE) // change of state
+            {
+                state = value;
+                if (value == STATE_ZOMBIE_DECIMATED)
+                {
+                    me->SetReactState(ReactStates::REACT_PASSIVE);
+                    me->AttackStop();
+                    me->SetTarget(ObjectGuid::Empty);
+                    // at this point, the zombie should be non attacking and non moving.
+
+                    me->SetWalk(true); // it doesnt seem to work with MoveFollow() (but it does work with MovePoint()). 
+                    //me->SetSpeed(UnitMoveType::MOVE_RUN, baseMoveSpeed[UnitMoveType::MOVE_WALK] / baseMoveSpeed[UnitMoveType::MOVE_RUN]); // Hack to stay in run mode but with walk speed in case using MoveFollow()
+
+                    timer = 1000;
+                }
+                else if (value == STATE_ZOMBIE_TOBE_EATEN)
+                {
+                    // loose aggro behavior
+                    me->SetReactState(ReactStates::REACT_PASSIVE);
+                    me->AttackStop();
+                    me->SetTarget(ObjectGuid::Empty);
+
+                    // and forced to stand still
+                    me->StopMoving();
+                    me->GetMotionMaster()->Clear();
+                    me->GetMotionMaster()->MoveIdle();
+                    me->ApplySpellImmune(0, IMMUNITY_MECHANIC, MECHANIC_GRIP, true); // not sure if this is blizz-like but this is very convenient
```

Comment:
Unsure about this. I'm not really sure what happens in retail when a zombie that is about to be eaten gets death griped.

if this line is remove, Gluth will follow the zombie.